inc/fw-update.php

```diff
@@ -702,3 +686,43 @@ function spbc_security_firewall_update_log($step)
 
     $spbc->save('fw_stats', true, false);
 }
+
+/**
+ * Decide need to force direct update
+ *
+ * @return bool
+ * @psalm-suppress NullArgument
+ */
+function spbc_security_firewall_update_is_switch_to_direct()
```

Comment:
What about returning the reason why the direct update started to write it to the log? Can we implement this?

Reply:
Good proposal, added https://github.com/CleanTalk/security-malware-firewall/pull/298/commits/ff0e1a92b0a75324a7b3693837a55d4d6622df04